test/domain_bridge/test_bidirectional_domain_bridge.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <string>
+
+#include "rclcpp/context.hpp"
+#include "rclcpp/node.hpp"
+#include "test_msgs/msg/basic_types.hpp"
+
+#include "domain_bridge/domain_bridge.hpp"
+
+#include "wait_for_publisher.hpp"
+
+class TestBidirectionalDomainBridge : public ::testing::Test
+{
+protected:
+  static void SetUpTestCase()
+  {
+    // Initialize contexts in different domains
+    context_1_ = std::make_shared<rclcpp::Context>();
+    rclcpp::InitOptions context_options_1;
+    context_options_1.auto_initialize_logging(false).set_domain_id(domain_1_);
+    context_1_->init(0, nullptr, context_options_1);
+
+    context_2_ = std::make_shared<rclcpp::Context>();
+    rclcpp::InitOptions context_options_2;
+    context_options_2.auto_initialize_logging(false).set_domain_id(domain_2_);
+    context_2_->init(0, nullptr, context_options_2);
+
+    node_options_1_.context(context_1_);
+    node_options_2_.context(context_2_);
+  }
+
+  static const std::size_t domain_1_{1u};
+  static const std::size_t domain_2_{2u};
+  static std::shared_ptr<rclcpp::Context> context_1_;
+  static std::shared_ptr<rclcpp::Context> context_2_;
+  static rclcpp::NodeOptions node_options_1_;
+  static rclcpp::NodeOptions node_options_2_;
+};
+
+const std::size_t TestBidirectionalDomainBridge::domain_1_;
+const std::size_t TestBidirectionalDomainBridge::domain_2_;
+std::shared_ptr<rclcpp::Context> TestBidirectionalDomainBridge::context_1_;
+std::shared_ptr<rclcpp::Context> TestBidirectionalDomainBridge::context_2_;
+rclcpp::NodeOptions TestBidirectionalDomainBridge::node_options_1_;
+rclcpp::NodeOptions TestBidirectionalDomainBridge::node_options_2_;
+
+TEST_F(TestBidirectionalDomainBridge, create_bidirectional_bridge)
```

Comment:
I don't feel strongly about it, but I think we can move this test to the [end to end](https://github.com/ros2/domain_bridge/blob/d706ca0c00e4f1a3b2d8028d1219ad76c1f01821/test/domain_bridge/test_domain_bridge_end_to_end.cpp) test suite and avoid re-writing the test fixture stuff.

Reply:
I agree, it does make more sense to have it in that file. Done in 359ea21